influxql/ast.go

```diff
@@ -137,6 +135,13 @@ type Statement interface {
 	RequiredPrivileges() ExecutionPrivileges
 }
 
+// HasDefaultDB provides an interface to get the default DB from a Statement.
+type HasDefaultDB interface {
```

Comment:
I'm not sure this is a great name for an interface, it seems rather different than other Go interfaces. It seems to me that it would be simpler if you just extended the statement interface itself such that it supported the method `DefaultDatabase()` which returned `""` is not default database was available.